include/bluetooth/l2cap.h

```diff
@@ -402,6 +456,25 @@ int bt_l2cap_chan_disconnect(struct bt_l2cap_chan *chan);
  *  Regarding to first input parameter, to get details see reference description
  *  to bt_l2cap_chan_connect() API above.
  *
+ *  When sending L2CAP data over an BR/EDR connection the application is sending
+ *  L2CAP PDUs. The application is required to have reserved
+ *  @ref BT_L2CAP_CHAN_SEND_RESERVE bytes in the buffer before sending.
+ *  The application should use the @def BT_L2CAP_BUF_SIZE helper to correctly
+ *  size the buffers for the for the outgoing buffer pool.
+ *
+ *  When sending L2CAP data over an LE connection the applicatios is sending
+ *  L2CAP SDUs. The application can optionally reserve
+ *  @ref BT_L2CAP_SDU_CHAN_SEND_RESERVE bytes in the buffer before sending.
+ *  By reserving bytes in the buffer the stack can use this buffer as a segment
+ *  directly, otherwise it will have to allocate a new segment for the first
+ *  segment.
```

Comment:
How come it is optional for LE? Seems slightly odd. 

Also, more importantly perhaps, what is the concrete side effect for the application if the application does not reserve the bytes? It says that the stack will have to allocate a new segment for the first, but why should the application care about that? Does it take a significant performance hit, and thus we recommend to reserve it, or something else?

Reply:
I think @Vudentz would be the right person to answer this. The implementation has not changed (AFACT), it's just being documented.

Reply:
For CoC the stack would check if there enough headroom and if there isn't it will allocate a buffer segment itself, so not there is no side effect except that it uses more buffers to segment the message. That sounds perfectly clear to me reading the documentation I wonder if that was not enough and we should rework it to make clear why it is not required to reserve headroom.